### PR TITLE
Added NeoScrypt Module

### PR DESCRIPTION
**Pull request by ahmedbodi**

### CODE REVIEW

**Comment by goodthebest** [comment]
@zone117x it seems neoscrypt patch but it's not yet merged with official package, any possibility in near future?

File: binding.gyp [hunk 1]
```diff
@@ -24,6 +24,7 @@
                 "sha1.c",
                 "x15.c",
                 "fresh.c",
+		"neoscrypt.c",
```
Comment (zone117x) [inline binding.gyp]: Minor whitespace issue - please fix

File: package.json [hunk 2]
```diff
@@ -26,6 +26,7 @@
         "blake",
         "shavite",
         "fugue",
-        "sha1"
+        "sha1",
+	"neoscrypt"
```
Comment (zone117x) [inline package.json]: And here.
Thanks @ahmedbodi you rock!

Comment (goodthebest) [reply package.json]: @ahmedbodi any progress on this pull request?

**Comment by ahmedbodi** [comment]
Wow! its been a long time since i did this, currently very busy with other stuff. But i'll fix and push in a few days

**Comment by ahmedbodi** [comment]
@zone117x All fixed bud, give it a whirl now

**Comment by HashUnlimited** [comment]
wow this might be the longest lasting PR I've ever seen, can we go ahead? :-D

**Comment by zone117x** [comment]
There are merge conflicts now. @HashUnlimited could you resolve them?

**Comment by aciddude** [comment]
Hey Guys, 

I've tried to implement this myself using the latest code as @ahmedbodi original fork is kinda out dated. 

I'm currently getting  rejected shares 

```
 nomp@nomp-01:~/node-open-mining-portal# node init.js 
2018-08-09 18:13:37 [POSIX]	[Connection Limit] (Safe to ignore) POSIX module not installed and resource (connection) limit was not raised
2018-08-09 18:13:37 [Master]	[CLI] CLI listening on port 17117
2018-08-09 18:13:38 [Master]	[PoolSpawner] Spawned 1 pool(s) on 1 thread(s)
2018-08-09 18:13:40 [Payments]	[feathercoin] Payment processing setup to run every 20 second(s) with daemon (feathercoinrpc@10.131.67.49:3080) and redis (127.0.0.1:6379)
2018-08-09 18:13:40 [Payments]	[feathercoin] Finished interval - time spent: 19ms total, 7ms redis, 11ms daemon RPC
(node-watch) First param in callback function  is replaced with event name since 0.5.0, use  `(evt, filename) => {}` if you want to get the filename
2018-08-09 18:13:40 [Website]	[Server] Website started on 127.0.0.1:80
2018-08-09 18:13:40 [Switching]	[Setup] (Thread 1) Loading last proxy state from redis
2018-08-09 18:13:40 [Pool]	[feathercoin] (Thread 1) Share processing setup with redis (127.0.0.1:6379)
2018-08-09 18:13:40 [Pool]	[feathercoin] (Thread 1) Network diff of 60.3279443070019 is lower than port 3008 w/ diff 512 and port 3256 w/ diff 256
2018-08-09 18:13:40 [Pool]	[feathercoin] (Thread 1) Stratum Pool Server Started for feathercoin [FTC] {neoscrypt}
						Network Connected:	Mainnet
						Detected Reward Type:	POW
						Current Block Height:	2305442
						Current Connect Peers:	9
						Current Block Diff:	63.503147344
						Network Difficulty:	60.3279443070019
						Network Hash Rate:	3.54 GH
						Stratum Port(s):	3008, 3032, 3256
						Pool Fee Percent:	1.5%
						Block polling every:	1000 ms
2018-08-09 18:13:40 [Switching]	[Setup] (neoscrypt) Setting proxy difficulties after pool start
2018-08-09 18:14:00 [Payments]	[feathercoin] Finished interval - time spent: 5ms total, 2ms redis, 3ms daemon RPC
2018-08-09 18:14:20 [Payments]	[feathercoin] Finished interval - time spent: 4ms total, 1ms redis, 4ms daemon RPC
2018-08-09 18:14:35 [Pool]	[feathercoin] (Thread 1) No new blocks for 55 seconds - updating transactions & rebroadcasting work
2018-08-09 18:14:40 [Payments]	[feathercoin] Finished interval - time spent: 4ms total, 2ms redis, 3ms daemon RPC
2018-08-09 18:14:55 [Pool]	[feathercoin] (Thread 1) Authorized 6x4LEQV88zRnBvZoH6ZNK6SeRxx4KiTyJs:x [127.0.0.1]
2018-08-09 18:14:55 [Pool]	[feathercoin] (Thread 1) Unknown stratum method from 6x4LEQV88zRnBvZoH6ZNK6SeRxx4KiTyJs [127.0.0.1]: mining.extranonce.subscribe
2018-08-09 18:15:00 [Payments]	[feathercoin] Finished interval - time spent: 4ms total, 1ms redis, 3ms daemon RPC
2018-08-09 18:15:05 [Pool]	[feathercoin] (Thread 1) Share rejected: {"job":"2","ip":"127.0.0.1","worker":"6x4LEQV88zRnBvZoH6ZNK6SeRxx4KiTyJs","difficulty":512,"error":"low difficulty share of 6.811827238496567e-10"}
``` 

my algoProperties.js for the node-stratum-pool

```
    'neoscrypt': {
        //diff: '000000fcff000000000000000000000000000000000000000000000000000000',
        //diff: '0x0000ffff00000000000000000000000000000000000000000000000000000000',
        muliplier: Math.pow(2, 5),
        hash: function(coinConfig){
            return function(){
                return multiHashing.neoscrypt.apply(this, arguments);
            }
        }
    },

```

is there anything else I need to change ?  just looking for some pointers so I can then send you a PR

**Comment by zone117x** [comment]
@hackmod want to take a stab at fixing up this PR?

**Comment by hackmod** [comment]
I guess this PR already done by @aciddude. https://github.com/zone117x/node-multi-hashing/pull/65

**Pull request closed without merging**